gandalf/src/main/java/io/github/btkelly/gandalf/holders/DialogStringsHolder.java

```diff
@@ -56,93 +57,88 @@ public String getUpdateAvailableTitle() {
     }
 
+    public void setUpdateAvailableTitle(@StringRes int updateAvailableTitle) {
+        this.updateAvailableTitle = context.getResources().getString(updateAvailableTitle);
+    }
+
+    public void setUpdateAvailableTitle(String updateAvailableTitle) {
+        this.updateAvailableTitle = updateAvailableTitle;
+    }
+
     @Nullable
     public String getUpdateAvailableMessage() {
         return updateAvailableMessage;
     }
 
+    public void setUpdateAvailableMessage(@StringRes int updateAvailableMessage) {
+        this.updateAvailableMessage = context.getResources().getString(updateAvailableMessage);
+    }
+
+    public void setUpdateAvailableMessage(@Nullable String updateAvailableMessage) {
+        this.updateAvailableMessage = updateAvailableMessage;
+    }
+
     public String getUpdateRequiredTitle() {
         if (updateRequiredTitle == null) {
             setUpdateRequiredTitle(R.string.update_required_title);
         }
         return updateRequiredTitle;
     }
 
-    @Nullable
-    public String getUpdateRequiredMessage() {
-        return updateRequiredMessage;
+    public void setUpdateRequiredTitle(@StringRes int updateRequiredTitle) {
+        this.updateRequiredTitle = context.getResources().getString(updateRequiredTitle);
     }
 
-    public String getAlertTitle() {
-        if (alertTitle == null) {
-            setUpdateAvailableTitle(R.string.alert_title);
-        }
-        return alertTitle;
+    public void setUpdateRequiredTitle(String updateRequiredTitle) {
+        this.updateRequiredTitle = updateRequiredTitle;
     }
 
     @Nullable
-    public String getAlertMessage() {
-        return alertMessage;
-    }
-
-    public String getDownloadUpdateButtonText() {
-        if (downloadUpdateButton == null) {
-            setDownloadUpdateButtonText(R.string.download_update_button);
-        }
-        return downloadUpdateButton;
+    public String getUpdateRequiredMessage() {
+        return updateRequiredMessage;
     }
 
-    public String getSkipUpdateButtonText() {
-        if (skipUpdateButton == null) {
-            setSkipUpdateButtonText(R.string.skip_update_button);
-        }
-        return skipUpdateButton;
+    public void setUpdateRequiredMessage(@StringRes int updateRequiredMessage) {
+        this.updateRequiredMessage = context.getResources().getString(updateRequiredMessage);
     }
 
-    public String getCloseAppButtonText() {
-        if (closeAppButton == null) {
-            setCloseAppButtonText(R.string.close_app_button);
-        }
-        return closeAppButton;
+    public void setUpdateRequiredMessage(@Nullable String updateRequiredMessage) {
+        this.updateRequiredMessage = updateRequiredMessage;
     }
 
-    public String getOkButtonText() {
-        if (okButton == null) {
-            setOkButtonText(R.string.ok_button);
+    public String getAlertTitle() {
+        if (alertTitle == null) {
+            setUpdateAvailableTitle(R.string.alert_title);
         }
```

Comment:
❓ why is this pattern different than e.g. `getUpdateRequiredMessage`?

Reply:
No idea 🤣 I didn't touch this class, it was all the code reformatter. Do you think it needs a change?

Reply:
Probably not?

Reply:
Same, if it isn't broke right?